File: frontend/src/api/foodData.ts
import client from './client';

export interface FoodDataItem {
  api_food_id: string;
  canonical_food_id?: string;
  display_name: string;
  canonical_name: string;
  description?: string;
  category?: string;
  is_fast_food: boolean;
  calories_kcal?: number;
  protein_g?: number;
  carbs_g?: number;
  sugars_g?: number;
  fiber_g?: number;
  fat_g?: number;
  saturated_fat_g?: number;
  sodium_mg?: number;
  cholesterol_mg?: number;
  quality_score?: number;
  variant_flavor?: string;
  variant_prep?: string;
  variant_protein?: string;
  canonical_food_name?: string;
  taxonomy_category?: string;
}

export interface RecipeNutrition {
  servings: number;
  total_calories: number;
  total_protein_g: number;
  total_carbs_g: number;
  total_fat_g: number;
  per_serving_calories: number;
  per_serving_protein_g: number;
  per_serving_carbs_g: number;
  per_serving_fat_g: number;
  ingredients: {
    name: string;
    matched: boolean;
    quantity: number;
    unit: string;
    calories_kcal?: number;
    protein_g?: number;
    carbs_g?: number;
    fat_g?: number;
  }[];
  matched_count: number;
  total_count: number;
}

export const foodDataApi = {
  search: (q: string, limit = 8) =>
    client.get<FoodDataItem[]>('/api/food-data/search', { params: { q, limit } }).then((r) => r.data),
  getOne: (id: string) =>
    client.get<FoodDataItem>(`/api/food-data/${id}`).then((r) => r.data),
  categories: () =>
    client.get<string[]>('/api/food-data/categories').then((r) => r.data),
  getRecipeNutrition: (recipeId: string) =>
    client.get<RecipeNutrition>(`/api/recipes/${recipeId}/nutrition`).then((r) => r.data),
};

File: backend/app/schemas/food_data.py
from typing import Optional
from pydantic import BaseModel


class CanonicalFoodOut(BaseModel):
    canonical_food_id: str
    canonical_name: str
    slug: str
    taxonomy_category: Optional[str] = None

    model_config = {"from_attributes": True}


class FoodDataOut(BaseModel):
    api_food_id: str
    canonical_food_id: Optional[str] = None
    display_name: str
    canonical_name: str
    description: Optional[str] = None
    category: Optional[str] = None
    is_fast_food: bool = False

    calories_kcal: Optional[float] = None
    protein_g: Optional[float] = None
    carbs_g: Optional[float] = None
    sugars_g: Optional[float] = None
    fiber_g: Optional[float] = None
    fat_g: Optional[float] = None
    saturated_fat_g: Optional[float] = None
    monounsaturated_fat_g: Optional[float] = None
    polyunsaturated_fat_g: Optional[float] = None
    water_g: Optional[float] = None
    sodium_mg: Optional[float] = None
    cholesterol_mg: Optional[float] = None
    quality_score: Optional[int] = None

    variant_flavor: Optional[str] = None
    variant_prep: Optional[str] = None
    variant_protein: Optional[str] = None

    # Canonical food name (from join) for better display
    canonical_food_name: Optional[str] = None
    taxonomy_category: Optional[str] = None

    model_config = {"from_attributes": True}


class FoodDataSearch(BaseModel):
    query: str
    category: Optional[str] = None
    limit: int = 20
